lib/seahorn/BvOpSem2TrackingRawMemMgr.cc

```diff
@@ -264,11 +256,24 @@ TrackingRawMemManager::memsetMetaData(MetadataKind kind, PtrTy ptr, Expr len,
   // make sure we can fit the supplied value in metadata memory slot
   assert(llvm::Log2_64(val) + 1 <= g_MetadataBitWidth &&
          "Metadata cannot fit!");
-  return MemValTy(
-      kind, memIn,
-      m_metadata_map[kind]->MemSet(ptr, m_ctx.alu().ui(val, g_MetadataBitWidth),
-                                   len, memIn.getMetadata(kind),
-                                   m_metadata_map[kind]->wordSzInBytes()));
+  auto index = static_cast<size_t>(kind) + 1;
```

Comment:
I am a bit confused for how `kind` and `memIn` are placed into `c`; perhaps this function can be more readable if all 3 elements in c are set explicitly.